### DISCUSSION TITLE
Noob question: I need to port forward, so the map is accessible to the WAN?

### DISCUSSION THREAD

**Discussion by philhzss** [Freika/dawarich#245] [Q&A] [answered]
I'm just wondering, is the only thing preventing anyone from accessing my location data the email/password I setup in Dawarich?

I probably did something wrong, because this doesn't feel very safe no? I setup Dawarich on a local machine, used Caddy to setup a reverse proxy, and in Owntracks I added the URL with the API key etc. It seems to work. I used another port than 80/443, because I already had a simple web server running on that port. So, I had to open the new port in the router and forward it to my machine running Dawarich.

But now, that means that from anywhere, I can connect to my home network and go to "/map" and get the login page. Is this normal? Is there a way to like.. allow Dawarich to receive location info from my phone on Owntracks without opening up access to the /map page? Or, is the login screen on Dawarich safe enough (as long as I choose a good password)?

Not sure if I'm doing something wrong or if this is normal, please let me know! Kind of new to this. Thank you!

**Comment by Freika** [comment, marked as answer]
I'd say it's safe enough, although security of your own setup is entirely on you, including the question "is it safe?".

I personally do not expose my homelab, including Dawarich, to the entire Internet for a number of reasons, and one of them is that I'm not proficient enough in security to do it in a most safe way possible. Also, to be honest, I find the question "I exposed it to Internet and it's available for everyone, is it safe?" a bit funny to say the least :)

Consider your stack as safe as its least safe element. Should I list examples of software that can become a vector of an attack?

**Reply by philhzss** [reply]
> Also, to be honest, I find the question "I exposed it to Internet and it's available for everyone, is it safe?" a bit funny to say the least :)


I agree, and that was the entire reason for this post; I felt that I did what I'm supposed to do to allow it to track my phone, and I felt that was a bad idea because it exposes it to the internet.. I was just asking if I had done something wrong but I guess this is the only way to get it to work without using a VPN. 

I also use home assistant so I'll try the method of using home assistant to push location updates to Dawarich. 

Thank you for your help!

**Reply by redge76** [reply]
Check Clouflare Access https://www.cloudflare.com/zero-trust/products/access/
There is tutorials everywhere. This is a very good way to expose you homelab with a pretty good security.
